src/main/java/org/example/springproject/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,21 @@
+package org.example.springproject.security;
+
+import lombok.RequiredArgsConstructor;
+import org.example.springproject.repository.user.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+
+        return userRepository.findByEmail(email).orElseThrow(
+                () -> new UsernameNotFoundException("Can't find user by email"));
```

Comment:
```suggestion
                () -> new EntityNotFoundException("Can't find user by email"));
```

add email to exception message